### PR TITLE
Add init option --coldcard-sd-path & PSBT exporting. Add broadcast subcommand

### PR DESCRIPTION
**Pull request by nickfarrow**
Rebased from xpub branch, keeping coldcard & PSBT related changes.

### CODE REVIEW

File: src/cmd/init.rs [hunk 1]
```diff
@@ -27,6 +27,9 @@ pub struct InitOpt {
     #[structopt(long, default_value = "12", name = "[12|24]")]
     /// The number of BIP39 seed words to use
     n_words: usize,
+    /// SD Card path for offline signing
+    #[structopt(long, parse(from_os_str))]
+    coldcard_sd_path: Option<PathBuf>,
```
Comment (nickfarrow) [inline src/cmd/init.rs:32]: `gun init --coldcard-sd-path /path/ bitcoin`

File: src/cmd/mod.rs [hunk 2]
```diff
@@ -520,7 +520,13 @@ pub fn decide_to_broadcast(
 
         if print_tx {
             Ok((
-                item! { "tx" => Cell::String(crate::hex::encode(&encode::serialize(&tx))) },
```
Comment (nickfarrow) [inline src/cmd/mod.rs:523]: Stop txid printing within massive table

File: src/cmd/wallet.rs [hunk 3]
```diff
@@ -231,6 +232,7 @@ pub struct SpendOpt {
 impl SpendOpt {
     pub fn spend_coins<D: BatchDatabase, Cs: CoinSelectionAlgorithm<D>, Ctx: TxBuilderContext>(
         self,
+        wallet_dir: &std::path::Path,
```
Comment (nickfarrow) [inline src/cmd/wallet.rs:235]: spend_coins now takes `wallet_dir` in order to load the config, and then save a PSBT onto SD card

File: src/cmd/wallet.rs [hunk 4]
```diff
@@ -270,34 +272,57 @@ impl SpendOpt {
             (psbt, vec![])
         };
 
-        party.wallet().sign(&mut psbt, SignOptions::default())?;
-
-        let finalized = party
```
Comment (nickfarrow) [inline src/cmd/wallet.rs:275]: We no longer care about `finalized`, but instead get different CmdOutput `output` based on each branch (currently coldcard or not).

File: src/cmd/wallet.rs [hunk 5]
```diff
@@ -554,5 +579,32 @@ pub fn run_split_cmd(wallet_dir: &std::path::Path, opt: SplitOpt) -> anyhow::Res
...
+}
+
+pub fn run_broadcast_cmd(
+    wallet_dir: &std::path::Path,
+    opt: BroadcastOpt,
+) -> anyhow::Result<CmdOutput> {
+    let BroadcastOpt { transaction_file } = opt;
+    let party = load_party(wallet_dir)?;
+    let contents =
+        fs::read_to_string(transaction_file).expect("Something went wrong reading the file");
```
Comment (LLFourn) [inline src/cmd/wallet.rs:599]: `?` with a .context.

Comment (LLFourn) [inline src/cmd/wallet.rs:599]: If the file name is `-` read from stdin.

**Review by LLFourn — COMMENTED** [review]
Made a few initial comments :)

**Comment by LLFourn** [comment]
superceded by #71

**Pull request closed without merging**